lib/services/history/base_history.ts

```diff
@@ -0,0 +1,34 @@
+import { PurePayload } from '@Payloads/pure_payload';
+import { ItemHistory } from '@Services/history/item_history';
+
+export type HistoryContent = {
+  itemUUIDToItemHistoryMapping: Record<string, ItemHistory>
```

Comment:
I think you over-extracted some things from SessionHistory.

Session history was meant to be an in-memory data store. So we had to use data structures to map uuids to item histories. This is not functionality we would want in an abstract class however. I'm actually not sure this abstract class is required at all tbh. If there are common functions we want, we should just define an `interface` instead.